File: jupyter2/problems/parametric/basics/jl/basics.jl
### begin my answer

### end my answer

function main()
    @assert(f(1) == 1)
    @assert(f("hello") == "hello")
    @assert(g(2) == [2])
    @assert(g("world") == ["world"])
    c0 = cell(0)
    @assert(get(c0) == 0)
    put!(c0, 42)
    c1 = cell("mimura")
    @assert(get(c1) == "mimura");
    put!(c1, "ohtake");
    @assert(get(c1) == "ohtake");
    @assert(h([]) == nothing)
    @assert(h([3]) == 3)
    @assert(h(["bye"]) == "bye")
    println("OK")
end

main()

File: jupyter2/problems/parametric/basics/rs/basics.rs
/** begin my answer */

/** end my answer */

fn main() {
    assert!(f(1) == 1);
    assert!(f("hello") == "hello");
    assert!(g(2) == [2]);
    assert!(g("world") == ["world"]);
    let mut c0 = Cell{x: 0};
    assert!(get(&c0) == &0);
    put(&mut c0, 42);
    let mut c1 = Cell{x: "mimura"};
    assert!(get(&c1) == &"mimura");
    put(&mut c1, "ohtake");
    assert!(get(&c1) == &"ohtake");
    assert!(h::<i64>(&vec![]) == None);
    assert!(h(&vec![3]) == Some(&3));
    assert!(h(&vec!["bye"]) == Some(&"bye"));
    println!("OK")
}
